controllers/provider-aws/pkg/controlplane/actuator.go

```diff
@@ -0,0 +1,359 @@
+	cpSecret, err := a.getSecret(cp.Namespace, common.CloudProviderSecretName)
+	if err != nil {
+		return err
+	}
+	senConfigMap, err := a.getConfigMap(cp.Namespace, common.CloudProviderConfigName)
+	if err != nil {
+		return err
+	}
+
+	// Compute checksums
+	csSecrets := util.MergeSecretMaps(deployedSecrets, map[string]*corev1.Secret{
+		common.CloudProviderSecretName: cpSecret,
+	})
+	csConfigMaps := map[string]*corev1.ConfigMap{
+		common.CloudProviderConfigName: senConfigMap,
+	}
+	checksums := util.ComputeChecksums(csSecrets, csConfigMaps)
+
+	// Collect additional CCM chart values
+	values = make(map[string]interface{})
+	if cpConfig.CloudControllerManager != nil {
+		values["featureGates"] = cpConfig.CloudControllerManager.FeatureGates
+	}
+
+	// Apply CCM chart
+	a.logger.Info("Applying CCM chart", "controlplane", objectName(cp), "chart", ccmChart.Name, "values", values)
+	err = ccmChart.Apply(ctx, a.gcs, a.chartApplier, cp.Namespace, cluster.Shoot, imagevector.ImageVector, checksums, values)
+	if err != nil {
+		return errors.Wrapf(err, "could not apply CCM chart for controlplane '%s'", objectName(cp))
+	}
+
+	return nil
+}
+
+// Delete reconciles the given controlplane and cluster, deleting the additional Shoot
+// control plane components as needed.
+func (a *actuator) Delete(
+	ctx context.Context,
+	cp *extensionsv1alpha1.ControlPlane,
+	cluster *extensionscontroller.Cluster,
+) error {
+	// Delete CCM objects
+	a.logger.Info("Deleting CCM objects", "controlplane", objectName(cp))
+	if err := ccmChart.Delete(ctx, a.client, cp.Namespace); err != nil {
+		return errors.Wrapf(err, "could not delete CCM objects for controlplane '%s'", objectName(cp))
+	}
+
+	// Delete config objects
+	a.logger.Info("Deleting configuration objects", "controlplane", objectName(cp))
+	if err := configChart.Delete(ctx, a.client, cp.Namespace); err != nil {
+		return errors.Wrapf(err, "could not delete configuration objects for controlplane '%s'", objectName(cp))
+	}
+
+	// Delete secrets
+	a.logger.Info("Deleting secrets", "controlplane", objectName(cp))
+	if err := controlPlaneSecrets.Delete(a.cs, cp.Namespace); err != nil {
+		return errors.Wrapf(err, "could not delete secrets for controlplane '%s'", objectName(cp))
+	}
+
+	return nil
+}
+
+func (a *actuator) getSecret(namespace, name string) (*corev1.Secret, error) {
+	secret, err := a.cs.CoreV1().Secrets(namespace).Get(name, metav1.GetOptions{})
+	if err != nil {
+		return nil, errors.Wrapf(err, "could not get secret '%s/%s'", namespace, name)
+	}
+	return secret, nil
+}
+
+func (a *actuator) getConfigMap(namespace, name string) (*corev1.ConfigMap, error) {
+	cm, err := a.cs.CoreV1().ConfigMaps(namespace).Get(name, metav1.GetOptions{})
+	if err != nil {
+		return nil, errors.Wrapf(err, "could not get configmap '%s/%s'", namespace, name)
+	}
+	return cm, nil
+}
+
+func objectName(obj runtime.Object) string {
+	k, _ := client.ObjectKeyFromObject(obj) // nolint:errcheck,gosec
```

Comment:
Is the metadata accessor https://github.com/gardener/gardener-extensions/blob/e884183001bedd48de979b6b21b3b03e53786b68/pkg/controller/predicate.go#L55-L60 an alternative?

Reply:
The idea is to get a string containing a namespaced name "<namespace>/<name>" from the object. It can be done in multiple ways, the one used here is from my perspective the most straightforward one. If you prefer, I can use the metadata accessor of course.